src/site/stages/build/index.js

```diff
@@ -38,6 +39,40 @@ const addSubheadingsIds = require('./plugins/add-id-to-subheadings');
 
 function defaultBuild(BUILD_OPTIONS) {
   const smith = Metalsmith(__dirname); // eslint-disable-line new-cap
+
+  // Override the normal use function to provide timing and logging information
+  smith._use = smith.use;
+  let stepCount = 0;
+  smith.use = function use(plugin, description) {
+    const step = ++stepCount;
+    if (!description) return smith._use(plugin);
+
+    let timerStart;
+
+    /* eslint-disable no-console */
+    return smith
+      ._use(() => {
+        console.log(chalk.cyan(`\nStep ${step} start: ${description}`));
+        timerStart = process.hrtime.bigint();
+      })
+      ._use(plugin)
+      ._use(() => {
+        const time = (process.hrtime.bigint() - timerStart) / 1000000n;
+
+        // Color the time
+        let color;
+        if (time < 1000) color = chalk.green;
+        else if (time < 10000) color = chalk.yellow;
+        else color = chalk.red;
+        const coloredTime = color(`[${time}ms]`);
+
+        console.log(
+          chalk.cyan(`Step ${step} end ${coloredTime}: ${description}`),
+        );
+      });
+    /* eslint-enable no-console */
+  };
```

Comment:
I love the use of `chalk`! Would honestly like to see us implement that and the `debug` module site-wide.

I know you already heard my case, but I again urge us to consider creating a standard plugin constructor that we can use, instead of an override on Metalsmith. I don't think there are that many risks involved with this, but it locks us into Metalsmith that much more, when our build is mostly custom at this point anyway. 

I also think a common constructor will make it easier for us to scale. For example, we need a method of passing the Drupal data from plugin to plugin if we ever want to break down `metalsmith-drupal` into manageable build steps. Right now, we do this with `BUILD_OPTIONS`, but that's really going outside of what that variable should be used for, so it isn't clear. I don't know how specifically yet, but as we rearchitect our build to align more with a traditional CMS, I think this could be pretty useful.

Reply:
Yeah, I'd be interested in that standard constructor. I'm not sure how we could make that, keep it simple, _and_ not increase our coupling with Metalsmith, but that does feel possible.

At any rate, I think it's beyond the scope of what I was trying to do here. This PR is just to get that build output in the shortest way possible. It should be easy to change how we go about that later if we decide to.